omegatrade-app/frontend/src/app/components/stock-dashboard/stock-dashboard.component.ts

```diff
@@ -0,0 +1,182 @@
+import { Component, OnInit } from '@angular/core';
+import { Chart, StockChart } from 'angular-highcharts';
+import { ActivatedRoute } from '@angular/router';
+import { RestService } from '../../services/rest.service';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { interval, Subscription } from "rxjs";
+import { take } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-stock-dashboard',
+  templateUrl: './stock-dashboard.component.html',
+  styleUrls: ['./stock-dashboard.component.css']
+})
+export class StockDashboardComponent implements OnInit {
+
+  selectedCompany: string = "";
+  lastUpdatedTime;
+  stock: StockChart;
+  stocks: any;
+  company: any;
+  companies: any;
+  loader: boolean = false;
+  subscription: Subscription;
+  intervalId: number;
+  sId: String;
+
+  constructor(private router: ActivatedRoute, private _snackBar: MatSnackBar, private restService: RestService) {
+  }
+
+  ngOnInit(): void {
+    this.sId = this.router.snapshot.queryParamMap.get('sId');
+    this.getCompanies();
+  }
+
+  getCompanies() {
+    this.restService.getData('companies/list')
+      .pipe(take(1))
+      .subscribe(
+        response => {
+          if (response && response.success) {
+            this.companies = response.data;
+            if (this.companies && this.companies.length > 0) {
+              if(this.sId){
+                this.selectedCompany = this.router.snapshot.queryParamMap.get('companyId');
+              }else{
```

Comment:
instead relying only on `this.sId` we can check companyId query param also

Reply:
`this.sId` is removed in the later commits as it is handled from backend